service/src/test/java/uk/gov/hmcts/reform/fpl/service/removeorder/CMORemovalActionTest.java

```diff
@@ -49,100 +52,88 @@ void isNotAcceptedIfAnyOtherClass() {
-                element(TO_REMOVE_ORDER_ID, CaseManagementOrder.builder().removalReason(REASON).build())
-            )
-        ));
-
+        assertThatThrownBy(() -> underTest.remove(caseData, caseDetailsMap, TO_REMOVE_ORDER_ID, cmoToRemove))
+            .usingRecursiveComparison()
+            .isEqualTo(unexpectedNumberOfCMOsRemovedException(TO_REMOVE_ORDER_ID, 2));
     }
 
     @Test
-    void shouldRemovedCaseManagementOrderWhenOtherOtherCMOisPresent() {
+    void shouldRemoveOrderWhenNoMatchingIDButMatchingHearingLabel() {
+        LocalDateTime differentStartDate = HEARING_START_DATE.plusDays(3);
+        CaseManagementOrder cmoToRemove = cmo(differentStartDate);
         CaseData caseData = CaseData.builder()
             .reasonToRemoveOrder(REASON)
-            .sealedCMOs(newArrayList(
-                element(TO_REMOVE_ORDER_ID, CaseManagementOrder.builder().build()),
-                element(ANOTHER_CASE_MANAGEMENT_ORDER_ID, CaseManagementOrder.builder().build())))
-            .hiddenCaseManagementOrders(null)
-            .hearingDetails(null)
+            .sealedCMOs(newArrayList(element(TO_REMOVE_ORDER_ID, cmoToRemove)))
+            .hiddenCaseManagementOrders(newArrayList(element(ALREADY_REMOVED_ORDER_ID, cmo())))
+            .hearingDetails(List.of(
+                element(HEARING_ID, hearing(CASE_MANAGEMENT_ORDER_ID, differentStartDate)),
+                element(ANOTHER_HEARING_ID, hearing(ANOTHER_CASE_MANAGEMENT_ORDER_ID))
+            ))
             .build();
 
         CaseDetailsMap caseDetailsMap = caseDetailsMap(CaseDetails.builder()
             .data(Map.of())
             .build());
 
-        underTest.remove(caseData, caseDetailsMap, TO_REMOVE_ORDER_ID, CaseManagementOrder.builder().build());
+        underTest.remove(caseData, caseDetailsMap, TO_REMOVE_ORDER_ID, cmoToRemove);
 
-        Map<String, List<?>> expectedData = new HashMap<>();
-        expectedData.put("hearingDetails", List.of());
-        expectedData.put("hiddenCaseManagementOrders", List.of(
-            element(TO_REMOVE_ORDER_ID, CaseManagementOrder.builder().removalReason(REASON).build())
-        ));
-        expectedData.put("sealedCMOs", List.of(
-            element(ANOTHER_CASE_MANAGEMENT_ORDER_ID, CaseManagementOrder.builder().build()))
+        assertThat(caseDetailsMap).isEqualTo(Map.of(
+            "hearingDetails", List.of(
+                element(HEARING_ID, hearing(null, differentStartDate)),
+                element(ANOTHER_HEARING_ID, hearing(ANOTHER_CASE_MANAGEMENT_ORDER_ID))
+            ),
+            "hiddenCaseManagementOrders", List.of(
+                element(ALREADY_REMOVED_ORDER_ID, cmo()),
+                element(TO_REMOVE_ORDER_ID, cmo(differentStartDate, REASON))
+            ))
         );
-
-        assertThat(caseDetailsMap).isEqualTo(expectedData);
     }
 
     @Test
-    void shouldRemovedCaseManagementOrderWhenNoHearing() {
```

Comment:
Old test is not possible, a case with a CMO would have to have at least one hearing